samples/bluetooth/ccp_call_control_client/src/main.c

```diff
@@ -0,0 +1,335 @@
+
+#include <zephyr/autoconf.h>
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/bluetooth/audio/tbs.h>
+#include <zephyr/bluetooth/audio/ccp.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gap.h>
+#include <zephyr/bluetooth/hci_types.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net_buf.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/util_macro.h>
+
+LOG_MODULE_REGISTER(ccp_call_control_client, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define SEM_TIMEOUT K_SECONDS(10)
+
+static struct bt_conn *peer_conn;
+/* client is not static as it is used for testing purposes */
+struct bt_ccp_call_control_client *client;
+static struct bt_ccp_call_control_client_bearers client_bearers;
+
+static K_SEM_DEFINE(sem_conn_state_change, 0, 1);
+static K_SEM_DEFINE(sem_security_updated, 0, 1);
+static K_SEM_DEFINE(sem_ccp_action_completed, 0, 1);
+
+static void connected_cb(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	(void)bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+	LOG_INF("Connected: %s", addr);
+
+	k_sem_give(&sem_conn_state_change);
+}
+
+static void disconnected_cb(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	if (conn != peer_conn) {
+		return;
+	}
+
+	(void)bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+	LOG_INF("Disconnected: %s (reason 0x%02x)", addr, reason);
+
+	bt_conn_unref(peer_conn);
+	peer_conn = NULL;
+	client = NULL;
+	memset(&client_bearers, 0, sizeof(client_bearers));
+	k_sem_give(&sem_conn_state_change);
+}
+
+static void security_changed_cb(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	if (err == 0) {
+		k_sem_give(&sem_security_updated);
+	} else {
+		LOG_ERR("Failed to set security level: %s(%u)", bt_security_err_to_str(err), err);
+	}
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected_cb,
+	.disconnected = disconnected_cb,
+	.security_changed = security_changed_cb,
+};
+
+static bool check_gtbs_support(struct bt_data *data, void *user_data)
+{
+	struct net_buf_simple svc_data;
+	bool *connect = user_data;
+	const struct bt_uuid *uuid;
+	uint16_t uuid_val;
+
+	if (data->type != BT_DATA_SVC_DATA16) {
+		return true; /* Continue parsing to next AD data type */
+	}
+
+	if (data->data_len < sizeof(uuid_val)) {
+		LOG_WRN("AD invalid size %u", data->data_len);
+		return true; /* Continue parsing to next AD data type */
+	}
+
+	net_buf_simple_init_with_data(&svc_data, (void *)data->data, data->data_len);
+
+	/* Pull the 16-bit service data and compare to what we are searching for */
+	uuid_val = net_buf_simple_pull_le16(&svc_data);
+	uuid = BT_UUID_DECLARE_16(sys_le16_to_cpu(uuid_val));
+	if (bt_uuid_cmp(uuid, BT_UUID_GTBS) != 0) {
+		/* We are looking for the GTBS service data */
+		return true; /* Continue parsing to next AD data type */
```

Comment:
I considered this comment unnecessary - but also nice for newcomers to better understand.

Reply:
Yeah, I think it makes sense in a sample

Reply:
Me too - and we should probably do more like this